### DISCUSSION TITLE
Testing machines that use `sendParent` in isolation

### DISCUSSION THREAD

**Discussion by chopfitzroy** [statelyai/xstate#2509] [General] [answered]
Heya,

I have two machines, a child machine called `serialMachine` and a parent parent called `menuMachine`.

The parent machine invokes the child machine like so:

```js
serial: {
  invoke: {
    id: 'serial-machine',
    src: serialMachine
  },
  on: {
    SERIAL_MACHINE_FAILURE: {
      target: 'failure',
      actions: ['setSerialFailureAction']
    },
    SERIAL_MACHINE_SUCCESS: {
      target: 'settings',
      actions: ['setSerialAction']
    }
  }
 }
```

And the child machine has the following actions:

```js
const sendParentFailureAction = sendParent(() => ({
    type: 'SERIAL_MACHINE_FAILURE',
}));

const sendParentCompletedAction = sendParent((context: SerialMachineContext) => ({
  type: 'SERIAL_MACHINE_SUCCESS',
  serial: context?.serial
}));
```

When I am testing the child machine for failure it the test fails with the following error:

```
 Missing onError handler and/or unhandled exception/promise rejection for invocation 'serial-machine.cache:invocation[0]'. Original error: 'Error: XXX'
 ```
 
 From what I can see this seems to be to do with using `sendParent` without being a child machine.
 
 If I change `sendParent` to `send` in the `sendParentFailureAction` function, the test passes but this breaks functionality.
 
 If I test the parent machine for failure it passes successfully and the child machine successfully sends the `SERIAL_MACHINE_FAILURE` to the parent.
 
 I believe this may be related to this [comment](https://github.com/statelyai/xstate/issues/1187#issuecomment-631139073) and the `event.data` being set to the error.
 
 Is there a way to mock the parent machine when trying to test the child in isolation?

**Comment by seloner** [comment, marked as answer]
You can override the setSerialFailureAction,setSerialAction
```js
const childmachine = // (your machine)

const mockChildMachine = childmachine.withConfig({
  actions: {
    setSerialFailureAction: (context, event) => {
      console.log('mock');
    },
 setSerialAction: (context, event) => {
      console.log('mock');
    }
  }
});
```

**Reply by chopfitzroy** [reply]
Hey @seloner this worked perfectly! I am so sorry for my delayed response on this.
